### DISCUSSION TITLE
Question about data security

### DISCUSSION THREAD

**Discussion by Nekronik** [vladmandic/human#121] [Q&A] [answered]
I tried to find if the library provides a statement about data security at any place, but I did not find any.

Due to the nature of our business, I need to confirm that the library is not leaking any information (system information, video/image snapshot, ...) to external endpoints.

I would love it if you clarified this.

Thanks.

**Comment by vladmandic** [comment, marked as answer]
`Human` library is fully self-contained and makes no external connections of any kind by itself unless user explicitly configures it to use externally hosted model weights (default is to load models from relative path, so no external connections are made) or to load images or video from an external URL (that is fully up to user). Note that even when weights are configured as externally hosted, they are just downloaded from the specified `modelPath`, library does not send or upload *any* data.

Library does not store or communicate any user or system data tracking, user provided inputs (images, video) or detection results.  
Additionally, library does not utilize any analytic services (such as *Google Analytics*).

Hope this helps & if there are any other specifics you need, let me know.

**Reply by Nekronik** [reply]
This is exaclty what I was looking for.
Thank you very much.
